code/go/0chain.net/smartcontract/dbs/event/updater.go

```diff
@@ -83,27 +119,27 @@ func (b *UpdateBuilder) AddUpdate(column string, values interface{}, expr ...str
 	}
 	b.unnests = append(b.unnests, fmt.Sprintf(UnnestTemplate, atype, column))
 
-	b.values = append(b.values, []interface{}{pq.Array(values)})
+	return b
+}
 
+func (b *UpdateBuilder) addToValues(values interface{}) *UpdateBuilder {
+	b.values = append(b.values, []interface{}{pq.Array(values)})
 	return b
 }
 
-type Query struct {
-	Q string
-	V []interface{}
+// Add condition in the form tableName.columnName = t.columnName
+func (b *UpdateBuilder) addWhereConditionFromValues(column string) *UpdateBuilder {
+	return b.addWhereCondition(b.tableName + "." + column, "=", "t." + column)
 }
 
-func (b *UpdateBuilder) build() *Query {
-	sets := ""
-	for _, s := range b.sets {
-		sets = sets + s
+func (b *UpdateBuilder) addWhereCondition(left, operator, right string) *UpdateBuilder {
+	if b.where != "" {
```

Comment:
please remove this if to more clear way of creating string, the first where will be added by JOIN clause I believe